src/moduleRoutes.js

```diff
@@ -68,3 +69,13 @@ function getModuleRoutes(stripes) {
 }
 
 export default getModuleRoutes;
+
+export const useModules = () => useContext(ModulesContext);
+
+export const useCurrentApp = () => {
+  const modules = useModules();
+  const location = useLocation();
+
+  const { app, settings } = modules;
+  return app.concat(settings).find(m => isQueryResourceModule(m, location));
```

Comment:
This looks great @zburke. Could we wrap this in `useMemo`?

Reply:
Hmmm. I did that, and then I read https://kentcdodds.com/blog/usememo-and-usecallback, and now I don't know if I should have done that 😕